app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,18 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.find_by(email: params.dig(:session, :email)&.downcase)
+    if user&.authenticate(params.dig(:session, :password))
```

Comment:
Toán tử Safe navigator `&.` hoạt động ntn?

Reply:
Dạ toán tử `&.` kiểm tra các giá trị có tồn tại hay không trước khi thực hiện các chức năng, Nếu biểu thức bên trái của `&.` (đối tượng) không phải là nil, thì toán tử sẽ gọi phương thức hoặc truy cập thuộc tính trên đối tượng đó như bình thường. Ngược lại, trả về nil ngay lập tức mà không gây ra lỗi